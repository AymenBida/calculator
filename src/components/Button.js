import PropTypes from 'prop-types';
import './button.css';

export default function Button(props) {
  const { name } = props;
  return (
    <button type="button" className="button">{ name }</button>
  );
}

Button.propTypes = {
  name: PropTypes.string.isRequired,
};
