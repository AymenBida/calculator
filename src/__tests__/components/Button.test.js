import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../../components/Button';

const handleClick = jest.fn();

it('renders correctly with the number "5"', () => {
  const tree = renderer.create(<Button name="5" clickHandler={handleClick} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with the operator "+"', () => {
  const tree = renderer.create(<Button name="+" clickHandler={handleClick} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with any string', () => {
  const tree = renderer.create(<Button name="aymen" clickHandler={handleClick} />).toJSON();
  expect(tree).toMatchSnapshot();
});
