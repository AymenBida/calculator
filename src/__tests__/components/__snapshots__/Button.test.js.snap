// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with any string 1`] = `
<button
  className="button"
  onClick={[Function]}
  onContextMenu={[Function]}
  type="button"
>
  aymen
</button>
`;

exports[`renders correctly with the number "5" 1`] = `
<button
  className="button"
  onClick={[Function]}
  onContextMenu={[Function]}
  type="button"
>
  5
</button>
`;

exports[`renders correctly with the operator "+" 1`] = `
<button
  className="button"
  onClick={[Function]}
  onContextMenu={[Function]}
  type="button"
>
  +
</button>
`;
